src/naive/datetime/mod.rs

```diff
@@ -533,6 +533,34 @@ impl NaiveDateTime {
         Some(NaiveDateTime { date, time })
     }
 
+    /// Adds given `Months` to the current date and time.
+    ///
+    /// Returns `None` when it will result in overflow.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    /// use chrono::{Months, NaiveDateTime};
+    /// let dt = |s| NaiveDateTime::from_str(s).unwrap();
+    /// let ca = |dt: NaiveDateTime, m: Months| dt.checked_add_months(m).unwrap();
+    /// let m = Months::new;
+    /// assert_eq!(ca(dt("2014-01-01T01:00:00"), m(1)), dt("2014-02-01T01:00:00"));
+    /// ```
+    ///
+    /// Overflow returns `None`.
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    /// use chrono::{Months, NaiveDateTime};
+    /// let dt = |s| NaiveDateTime::from_str(s).unwrap();
+    /// let m = Months::new(core::i32::MAX as u32 + 1);
+    /// assert_eq!(dt("2014-01-01T01:00:00").checked_add_months(m), None);
+    /// ```
+    pub fn checked_add_months(self, rhs: Months) -> Option<NaiveDateTime> {
+        self.date().checked_add_months(rhs).map(|dt| dt.and_time(self.time()))
```

Comment:
I think this can be simplified to

```rust
self.date = self.date.checked_add_months(rhs)?;
```

I definitely think we should not be using the `date()` and `time()` accessors here.